src/main/java/io/github/jelastic/core/repository/ElasticRepository.java

```diff
@@ -320,6 +325,46 @@ public void reAlias(@NotEmpty  String newIndex, @NotEmpty String aliasName) {
         return ElasticUtils.getResponse(multiGetItemResponses, idSearchRequest.getKlass());
     }
 
+    /**
+     * Load data of entire index based on batchSize using ES scroll API
+     * @param <T> Response class Type
+     * @param index Index to be loaded
+     * @param query jelastic query object
+     * @param batchSize Will tell ElasticClient the size of each fetch, should be <= 10000 for better performance
+     * @return List<T> list of all objects in that index
+     */
+    public <T> List<T> loadAll(String index, Query query, Integer batchSize, Class<T> klass) {
+        final Scroll scroll = new Scroll(TimeValue.timeValueMinutes(1L));
+
+        QueryBuilder queryBuilder = queryManager.getQueryBuilder(query);
+        SearchRequestBuilder searchRequestBuilder = elasticClient
+                .getClient()
+                .prepareSearch(index)
+                .setQuery(queryBuilder)
+                .setSize(batchSize)
+                .setScroll(scroll);
+        SearchResponse searchResponse = searchRequestBuilder
+                .execute()
+                .actionGet();
+
+        String scrollId = searchResponse.getScrollId();
+        List<T> batchedResult = ElasticUtils.getResponse(searchResponse, klass);
```

Comment:
How many results can this searchResponse contain? ElasticUtils.getResponse does casting, this is going to be really expensive, are you going to impose any size checks on this scroll method?

Reply:
1. It fetches in batches of 10k till all the data is read from ES
2. No size check in place as of now. It is unbounded. In our use case we are caching entire ES data into In-Memory cache. Which might hit 100k record at max.

Reply:
You might want to still impose a size check, this lib is NOT for your use-case, alone! It is a general purpose lib, and I would like for size check to be imposed from config, with a default to 10K.

Reply:
Added a configuration for maxResultSize, defaulting to 10k.

Reply:
Please don't do an assert, please use the JElasticException. And use that as a guard rail, please use the constant from the configuration in the comparator not a hardcoded value!

Reply:
Added exception if limit is reached.

Reply:
What kind of coding is that? Why would you do that inside the while loop? A guard rail is supposed to work at the beginning of a method. Pl. make sure you do it before you do anything else. Pl. understand why you are throwing this exception - this is to mean, look I breached my size so I'm not performing this action - which is the narrative! I don't see your code doing it. Pl. think and write code!

Reply:
How would I know if the fetched result is exceeding a specified limit unless I read the data from ES ?

Reply:
Thanks for making the change. The idea of guard rail is to protect you from things going wrong, not as a reactive check.

Reply:
Thanks for making the change. The idea of a guard rail is to protect you from doing any wrong, not to check reactively. This check now works.